script.js

```diff
@@ -1,4 +1,109 @@
+      playerRolls.push(roll);
+      console.log(`${roll}`);
+    }
+    roundCounter = `player1Chooses`;
+    var rollText =
+      `Welcome Player 1<br><br>` + rollSentence(playerRolls[0], playerRolls[1]);
+    // doesn't work: return `Welcome Player 1.<br><br>${rollSentence}`;
+    // doesn't work: var rollSentence = rollSentence (playerRolls[0], playerRolls[1])
+    // doesn't work: return `Welcome Player 1.<br><br>` + rollSentence;
+    return rollText;
+  }
+  if (roundCounter == `player1Chooses`) {
+    if (input != 1 && input != 2) {
+      var errorText =
+        rollSentence(playerRolls[0], playerRolls[1]) +
+        `<br><br>Please choose "1" or "2"!`;
+      return errorText;
+    }
+    if (input == 1) {
+      player1Choice = Number(`${playerRolls[0]}${playerRolls[1]}`);
+    }
+    if (input == 2) {
+      player1Choice = Number(`${playerRolls[1]}${playerRolls[0]}`);
+    }
+    playerRolls = [];
+    roundCounter = `player2Rolls`;
+    return `Player 1 chose ${player1Choice}.<br><br>Player 2 it is your turn`;
+  }
+  if (roundCounter == `player2Rolls`) {
+    for (var counter = 0; counter < 2; counter += 1) {
+      var roll = dice[Math.floor(Math.random() * dice.length)];
+      playerRolls.push(roll);
+      console.log(`${roll}`);
+    }
+    roundCounter = `player2Chooses`;
+    var rollText =
+      `Welcome Player 2<br><br>` + rollSentence(playerRolls[0], playerRolls[1]);
+    // doesn't work: return `Welcome Player 1.<br><br>${rollSentence}`;
+    // doesn't work: var rollSentence = rollSentence (playerRolls[0], playerRolls[1])
+    // doesn't work: return `Welcome Player 1.<br><br>` + rollSentence;
+    return rollText;
+  }
+  if (roundCounter == `player2Chooses`) {
+    if (input != 1 && input != 2) {
+      var errorText =
+        rollSentence(playerRolls[0], playerRolls[1]) +
+        `<br><br>Please choose "1" or "2"!`;
+      return errorText;
+    }
+    if (input == 1) {
+      player2Choice = Number(`${playerRolls[0]}${playerRolls[1]}`);
+    }
+    if (input == 2) {
+      player2Choice = Number(`${playerRolls[1]}${playerRolls[0]}`);
+    }
+    playerRolls = [];
+    roundCounter = `player1Rolls`;
+    if (player1Choice > player2Choice) {
+      player1Wins += 1;
+      var outputText =
+        choiceSentence(player2Choice, player1Choice) +
+        `<br><br>` +
+        scoreSentence(player1Wins, player2Wins, draws);
+      return outputText;
+    }
+    if (player1Choice < player2Choice) {
+      player2Wins += 1;
+      var outputText =
+        choiceSentence(player2Choice, player1Choice) +
+        `<br><br>` +
+        scoreSentence(player1Wins, player2Wins, draws);
+      return outputText;
+    }
+    if (player1Choice == player2Choice) {
+      draws += 1;
+      var outputText =
+        choiceSentence(player2Choice, player1Choice) +
+        `<br><br>` +
+        scoreSentence(player1Wins, player2Wins, draws);
+      return outputText;
+    }
+  }
+};
```

Comment:
again a bit repetitive, see that the only diff is whether we increment player1Wins, player2Wins or draws? why not have the outputText declaration outside, so that you only need to declare it once

Reply:
not sure what you mean in your desired specific feedback but i believe you're referring to this part? example of what i mean:

```
    if (player1Choice > player2Choice) {
      player1Wins += 1;
    } else if (player1Choice < player2Choice) {
      player2Wins += 1;
    } else {
      draws += 1;
    }
   var outputText =
        choiceSentence(player2Choice, player1Choice) +
        `<br><br>` +
        scoreSentence(player1Wins, player2Wins, draws);
  return outputText;
  ```